src/hooks/__tests__/useNavigatorOnline.spec.tsx

```diff
@@ -0,0 +1,44 @@
+import { act, renderHook } from '@testing-library/react';
+import { useNavigatorOnline } from '../custom-hooks';
+
+describe('useNavigatorOnline', () => {
+    it('initializes as true when browser is online', () => {
+        const { result } = renderHook(() => useNavigatorOnline());
+        expect(result.current).toBe(true);
+    });
+
+    it('sets status to false when offline event is triggered', () => {
+        const { result } = renderHook(() => useNavigatorOnline());
+
+        act(() => {
+            window.dispatchEvent(new Event('offline'));
+        });
+
+        expect(result.current).toBe(false);
+    });
+
+    it('sets status to true when online event is triggered', () => {
+        Object.defineProperty(navigator, 'onLine', {
```

Comment:
nit:
```suggestion
        Object.defineProperty(navigator, 'online', {
```

Reply:
The navigator api has written online like this
![Screenshot 2024-06-18 at 11 57 40 AM](https://github.com/deriv-com/p2p/assets/93518187/24750c41-61ca-4be2-8f23-0a6e57272d0a)

Reply:
oooo 👍